pkg/cmd/server/origin/master_config.go

```diff
@@ -675,11 +675,20 @@ func newAdmissionChain(pluginNames []string, admissionConfigFilename string, plu
 				return nil, err
 			}
 
-			plugin, err = admissionregistry.OriginAdmissionPlugins.InitPlugin(pluginName, pluginConfigReader, admissionInitializer)
+			plugin = nil
+			enabled, pluginConfigReaderCopy, err := admissionregistry.IsAdmissionPluginEnabled(pluginName, pluginConfigReader)
 			if err != nil {
-				// should have been caught with validation
 				return nil, err
 			}
+
+			if enabled {
```

Comment:
`if !enabled {continue}` to reduce nesting?

Reply:
Why to send disabled plugin all the way down when what we get in return is just plugin as nil.

Reply:
> Why to send disabled plugin all the way down when what we get in return is just plugin as nil.

I'm trying to see why its important to skip this and return in the next if instead of eliminating this nesting in keeping with golang style.